### PR TITLE
Added pretty? option to POST api/dataset/native route

### LINKED ISSUE #28801: Display Generated SQL in a nicely formatted, ideally syntax-colored way
When you click on "convert question to SQL", the default sql is not formatted and creates one long line. This makes it very difficult to parse and understand the question when you need to be in SQL to modify. It would be nice if it were nicely formatted.

Seems like something where we could slap a library and call it a day.

**How important is this feature to you?**
It would make it a lot easier to understand the generated SQL, sanity check it, and debug it for SQL writers, who are an influential and important bunch.

### PR DESCRIPTION
**Pull request by markbastian**
Added an opt-in body param of `pretty?` which will format the native SQL command if true. The original non-formatted SQL will be returned if the flag is false or the formatter throws an exception.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29336)

### CODE REVIEW

File: src/metabase/api/dataset.clj [hunk 1]
```diff
@@ -157,10 +159,18 @@
 #_{:clj-kondo/ignore [:deprecated-var]}
 (api/defendpoint-schema POST "/native"
   "Fetch a native version of an MBQL query."
-  [:as {query :body}]
+  [:as {{:keys [database pretty?] :as query} :body}]
+  {database s/Int
```
Comment (tsmacdonald) [inline src/metabase/api/dataset.clj:163]: nbd, but we normally use `ms/PositiveInt`

Comment (tsmacdonald) [reply src/metabase/api/dataset.clj:163]: er, this is plumatic, not malil, so `su/IntGreaterThanZero`. But it'd be nice to convert it to a malli defendpoint

```
{database ms/IntGreaterThanZero
 pretty?  [:maybe :boolean]}
 ```

File: src/metabase/api/dataset.clj [hunk 2]
```diff
@@ -157,10 +159,18 @@
...
+   pretty?  (s/maybe s/Bool)}
   (binding [persisted-info/*allow-persisted-substitution* false]
     (qp.perms/check-current-user-has-adhoc-native-query-perms query)
-    (qp/compile-and-splice-parameters query)))
+    (let [{q :query :as compiled} (qp/compile-and-splice-parameters query)
+          driver (driver.u/database->driver database)
+          q'     (if pretty?
```
Comment (tsmacdonald) [inline src/metabase/api/dataset.clj:169]: s/`q'`/`formatted-query`?

File: src/metabase/api/dataset.clj [hunk 3]
```diff
@@ -157,10 +159,18 @@
...
+                   (try (mdb.query/format-sql q driver)
+                        (catch Exception _e q))
```
Comment (tsmacdonald) [inline src/metabase/api/dataset.clj:171]: Another option is

```
(or (u/ignore-exceptions (mdb.query/format-sql q driver))
    q)
```

but I'm not sure that's actually any better :upside_down_face:

**Review by tsmacdonald — APPROVED** [review]
Do as you will with my nitpicking, it looks fine to me :boat:

File: src/metabase/api/dataset.clj (hunk 2 above)
Comment (markbastian) [reply src/metabase/api/dataset.clj:169]: Works for me

File: src/metabase/api/dataset.clj (hunk 1 above)
Comment (markbastian) [reply src/metabase/api/dataset.clj:163]: Good idea.

File: src/metabase/api/dataset.clj (hunk 3 above)
Comment (markbastian) [reply src/metabase/api/dataset.clj:171]: I do like that better.

File: test/metabase/api/dataset_test.clj [hunk 4]
```diff
@@ -303,7 +304,21 @@
...
+                           "  \"PUBLIC\".\"VENUES\".\"NAME\" AS \"NAME\"\n"
+                           "FROM\n"
+                           "  \"PUBLIC\".\"VENUES\"\n"
+                           "LIMIT\n"
+                           "  1048575")
+              :params nil}
+             (mt/user-http-request :rasta :post 200 "dataset/native"
+                                   (assoc
+                                    (mt/mbql-query venues {:fields [$id $name]})
+                                     :pretty? true)))))))
```
[the remark this answers is not in the document]
Comment (markbastian) [reply test/metabase/api/dataset_test.clj]: Done

**Pull request merged**